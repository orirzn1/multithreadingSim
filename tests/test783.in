L9
S6
O8
N8

T0
I@0x00000000
ADD $0, $4, $4
SUB $0, $1, $1
LOAD $4, $1, 0x2
ADDI $3, $1, 0x2
HALT $0

T1
I@0x00000000
ADDI $0, $0, 0x4
STORE $1, $5, $1
LOAD $0, $6, $3
SUB $3, $4, $7
HALT $0

T2
I@0x00000000
STORE $5, $1, $4
SUBI $2, $0, 0x4
LOAD $2, $6, $4
ADDI $5, $3, 0x2
SUB $7, $2, $7
STORE $5, $4, 0x4
STORE $5, $3, 0x1
SUBI $1, $2, 0x2
STORE $7, $3, $2
HALT $0

T3
I@0x00000000
ADD $7, $4, $7
ADDI $2, $3, 0x3
SUBI $3, $0, 0x3
ADD $0, $6, $4
HALT $0

T4
I@0x00000000
ADDI $6, $2, 0x1
SUB $2, $0, $3
ADD $4, $3, $6
ADDI $0, $6, 0x2
ADDI $5, $4, 0x1
SUBI $6, $2, 0x4
LOAD $7, $7, $4
SUB $4, $6, $7
HALT $0

T5
I@0x00000000
STORE $3, $5, $5
SUBI $7, $3, 0x2
ADD $4, $4, $5
ADDI $3, $2, 0x1
SUBI $7, $7, 0x4
SUB $0, $3, $1
LOAD $7, $3, $2
ADDI $5, $7, 0x1
HALT $0

T6
I@0x00000000
SUB $7, $5, $4
SUB $5, $7, $4
SUBI $3, $3, 0x1
STORE $4, $2, $7
ADD $2, $4, $4
SUBI $0, $7, 0x1
LOAD $6, $2, 0x1
ADD $2, $4, $3
LOAD $4, $3, 0x1
HALT $0

T7
I@0x00000000
SUB $5, $3, $5
SUBI $7, $7, 0x3
SUBI $1, $5, 0x3
SUB $4, $0, $2
STORE $2, $4, 0x0
SUB $6, $0, $4
ADD $3, $2, $5
ADDI $5, $1, 0x1
HALT $0

D@0x0
0x7
0x7
0x7
0x2
0x6
