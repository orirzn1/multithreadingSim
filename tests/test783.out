
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x2	R4 = 0x7	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0xFFFFFFFE	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x2	R6 = 0x0	R7 = 0x0
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0x3	R3 = 0xFFFFFFFD	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 4:
	R0 = 0x3	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x2	R6 = 0xFFFFFFFC	R7 = 0x0
Register file thread id 5:
	R0 = 0x1	R1 = 0x0	R2 = 0x0	R3 = 0x1	R4 = 0x0	R5 = 0x1	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0xFFFFFFFF	R1 = 0x0	R2 = 0xFFFFFFFF	R3 = 0xFFFFFFFF	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 7:
	R0 = 0x0	R1 = 0xFFFFFFFD	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFE	R6 = 0x0	R7 = 0xFFFFFFFD
Blocked MT CPI for this program 3.806452

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x2	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0xFFFFFFFE	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x2	R6 = 0x0	R7 = 0x0
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0x3	R3 = 0xFFFFFFFD	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 4:
	R0 = 0x3	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x2	R6 = 0xFFFFFFFC	R7 = 0x0
Register file thread id 5:
	R0 = 0x1	R1 = 0x0	R2 = 0x0	R3 = 0x1	R4 = 0x0	R5 = 0x1	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0xFFFFFFFF	R1 = 0x0	R2 = 0xFFFFFFFF	R3 = 0xFFFFFFFF	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 7:
	R0 = 0x0	R1 = 0xFFFFFFFD	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFE	R6 = 0x0	R7 = 0xFFFFFFFD
Finegrained Multithreading CPI for this program 1.161290

